driver-core/src/main/java/com/datastax/driver/core/PagingState.java

```
@@ -0,0 +1,158 @@
+        hashToCheck.get(original);
+        toTest = this.generatePagingStateHash(statement);
+
+        return Arrays.equals(original, toTest);
+    }
+
+    /**
+     * Return the hash digest generated after the creation of the
+     * PagingState object.
+     *
+     * @return the MD5 hash generated wrapped in a {@link java.nio.ByteBuffer}.
+     */
+    public ByteBuffer getHashDigest() {
+        return this.hashDigest;
+    }
+
+    /**
+     * Return the paging state buffer returned on the result of a query.
+     *
+     * @return the paging state buffer in a {@link java.nio.ByteBuffer} object.
+     */
+    public ByteBuffer getRawState() {
+        return this.pagingState;
+    }
+
+    /**
+     * Export the paging state and its MD5 hash in a string with the
+     * two values separated by a "_".
+     *
+     * @return the string of the two values separated by a "_".
+     */
+    @Override
+    public String toString() {
+        return Bytes.toRawHexString(this.pagingState) + "_" + Bytes.toRawHexString(this.hashDigest);
+    }
+
+    /**
+     * Create a PagingState object from a string previously saved of a PagingState
+     * object. As for the construction of a paging state object, the values must match
+     * the same statement (and potential parameters) since a hash verification is going
+     * to be made from the string.
+     *
+     * @param string the string value.
+     * @return the PagingState object created. Note that if the verification fails, null
+     * is returned.
+     */
+    public static PagingState fromString(String string) {
+        String[] splits = string.split("_");
+        if (splits.length == 2) {
+            ByteBuffer bbPagingState = ByteBuffer.wrap(Bytes.fromRawHexString(splits[0], 0));
+            ByteBuffer bbHash = ByteBuffer.wrap(Bytes.fromRawHexString(splits[1], 0));
+            return new PagingState(bbPagingState, bbHash);
+        } else
+            return null;
```

We should throw `InvalidPagingStateException`. And also in the cases where `fromRawHexString` throws an `IllegalArgumentException`.